Code/Source/Frame/ROS2Transform.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+*/
+
+#include "Frame/ROS2Transform.h"
+#include "Utilities/ROS2Conversions.h"
+#include "ROS2/ROS2Bus.h"
+
+namespace ROS2
+{
+    namespace Internal
+    {
+        class StaticTransformPublisher : public TransformPublisher
+        {
+        public:
+            void Publish(const geometry_msgs::msg::TransformStamped& transformMessage) override
+            {
+                if (m_published)
+                    return; // Consider adding a warning here as well
+
+                return ROS2Interface::Get()->BroadcastStaticTransform(transformMessage);
+                m_published = true;
+            }
```

Comment:
This code will warn on clang and and because O3DE compilers with warrnings as errors on, this will trigger an error when using this gem in O3DE with the clang compiler.

The `m_published = true` needs to be before the return statement.

Reply:
Good catch. I am compiling with clang and somehow I did not get any warning. How is this possible - I will look into it.